packages/components/src/components/Select/Select.tsx

```diff
@@ -0,0 +1,186 @@
+import { Box } from "../../primitives/Box";
+import { Icon } from "../Icon";
+import { SelectItem } from "./SelectItem";
+import type { SelectItemProps } from "./SelectItem";
+import { SelectPopover } from "./SelectPopover";
+
+export interface SelectProps
+  extends Omit<
+    SelectPrimitiveProps,
+    | "accessibleWhenDisabled"
+    | "autoFocus"
+    | "clickOnEnter"
+    | "clickOnSpace"
+    | "focusable"
+    | "moveOnKeyDown"
+    | "nonce"
+    | "onFocusVisible"
+    | "showOnKeyDown"
+    | "state"
+    | "toggleOnClick"
+    | "toggleOnPress"
+    | "typeahead"
+    | "wrapElement"
+  > {
+  /** The default selected option of the select. */
+  defaultValue?: string;
+  /** Sets the state of the select to disabled, so a user is not able to interact with it. */
+  disabled?: boolean;
+  /** Sets the state of the select to an error state. */
+  hasError?: boolean;
+  /** The `id` of the select. */
+  id?: string;
+  /** The options for the select. */
+  items: SelectItemProps[];
+  /** The `name` of the select. */
+  name?: string;
+  /** Sets the select state to required, so a user has to provide a value in order to be valid. */
+  required?: boolean;
+  /** Function that will be called when setting the select value state. */
+  setValue?: SelectStateProps["setValue"];
+  /** The selected option of the select. */
+  value?: string;
+}
+
+const getSelectStyles = (
+  disabled?: boolean,
+  hasError?: boolean
+): {
+  backgroundColor: SystemProp<keyof Theme["colors"], Theme> | "transparent";
+  borderColor: SystemProp<keyof Theme["colors"], Theme>;
+} => {
+  if (disabled) {
+    return {
+      backgroundColor: "colorBackgroundWeak",
+      borderColor: "colorBorderWeaker",
+    };
+  }
+  if (hasError) {
+    return {
+      backgroundColor: {
+        _: "colorBackground",
+        hover: "colorBackgroundWeakest",
+      },
+      borderColor: "colorBorderError",
+    };
+  }
+  return {
+    backgroundColor: {
+      _: "colorBackground",
+      hover: "colorBackgroundWeakest",
+    },
+    borderColor: "colorBorderWeaker",
+  };
+};
+
+/** A select is an styled dropdown form element that allows users to select a value from a list. */
+const Select = React.forwardRef<HTMLButtonElement, SelectProps>(
+  (
+    {
+      defaultValue,
+      disabled,
+      hasError,
+      id,
+      items,
+      name,
+      required,
+      setValue,
+      value,
+      ...props
+    },
+    ref
+  ) => {
+    const select = useSelectState({
+      defaultValue,
+      value,
+      sameWidth: true,
+      gutter: 4,
+      setValue,
+    });
+
+    return (
+      <>
+        <Box.button
+          alignItems="center"
+          appearance="none"
+          aria-required={required}
+          as={SelectPrimitive}
+          background="none"
+          borderRadius="borderRadius20"
+          borderStyle="borderStyleSolid"
+          borderWidth="borderWidth10"
+          color="colorText"
```

Comment:
I think it is almost the same as what Brian asked 😅 but, 

![image](https://user-images.githubusercontent.com/5320963/207294811-96ae1b2f-c6e3-4614-b2ed-ed0c617e121a.png)

Shouldn't this color be strongest when the value was selected? Having it in this way, it seems like a placeholder. 🤔 

What do you think?

Reply:
@BrianLincoln @apvale I worked with design on this a bit more and came up with a solution for an initial select option since selects don't really have placeholders. Its a `initial` boolean prop that can be added to a SelectItem. It mainly adjust styling, but could also be used for form validation down the line. We'll need to test that a bit more in an actual use-case in the app in the future.